test/unit/auth/auth-api-request.spec.ts

```diff
@@ -3549,6 +3550,552 @@ AUTH_REQUEST_HANDLER_TESTS.forEach((handler) => {
+          return requestHandler.listTenants(maxResults, nextPageToken)
+            .then((resp) => {
+              throw new Error('Unexpected success');
+            }, (error) => {
+              expect(error).to.deep.equal(expectedError);
+              expect(stub).to.have.been.calledOnce.and.calledWith(callParams(path, method, data));
+            });
+        });
+      });
+
+      describe('deleteTenant', () => {
+        const path = '/v2beta1/projects/project_id/tenants/tenant_id';
+        const method = 'DELETE';
+        const tenantId = 'tenant_id';
+        const expectedResult = utils.responseFrom({});
+
+        it('should be fulfilled given a valid tenant ID', () => {
+          const stub = sinon.stub(HttpClient.prototype, 'send').resolves(expectedResult);
+          stubs.push(stub);
+
+          const requestHandler = handler.init(mockApp) as AuthRequestHandler;
+          return requestHandler.deleteTenant(tenantId)
+            .then((result) => {
+              expect(result).to.be.undefined;
+              expect(stub).to.have.been.calledOnce.and.calledWith(callParams(path, method, {}));
+            });
+        });
+
+        const invalidTenantIds = [null, NaN, 0, 1, true, false, '', ['tenant_id'], [], {}, { a: 1 }, _.noop];
+        invalidTenantIds.forEach((invalidTenantId) => {
+          it('should be rejected given an invalid tenant ID:' + JSON.stringify(invalidTenantId), () => {
+            const expectedError = new FirebaseAuthError(AuthClientErrorCode.INVALID_TENANT_ID);
+
+            const requestHandler = handler.init(mockApp) as AuthRequestHandler;
+            return requestHandler.deleteTenant(invalidTenantId as any)
+              .then((result) => {
+                throw new Error('Unexpected success');
+              }, (error) => {
+                expect(error).to.deep.equal(expectedError);
+              });
+          });
+        });
+
+        it('should be rejected given a backend error', () => {
+          const expectedError = new FirebaseAuthError(AuthClientErrorCode.TENANT_NOT_FOUND);
+          const expectedServerError = utils.errorFrom({
+            error: {
+              message: 'TENANT_NOT_FOUND',
+            },
+          });
+          const stub = sinon.stub(HttpClient.prototype, 'send').rejects(expectedServerError);
+          stubs.push(stub);
+
+          const requestHandler = handler.init(mockApp) as AuthRequestHandler;
+          return requestHandler.deleteTenant(tenantId)
+            .then((resp) => {
+              throw new Error('Unexpected success');
+            }, (error) => {
+              expect(error).to.deep.equal(expectedError);
+              expect(stub).to.have.been.calledOnce.and.calledWith(callParams(path, method, {}));
+            });
+        });
+      });
+
+      describe('createTenant', () => {
+        const path = '/v2beta1/projects/project_id/tenants';
+        const method = 'POST';
```

Comment:
Nit: Perhaps call it `postMethod` for clarity.

Reply:
done